### ISSUE TITLE
[Support Question] Generalized Slowness

### ISSUE THREAD

**Issue by nrherron92** [gocodebox/lifterlms#2036]
### Reproduction Steps

+ We've reached a point where we don't know how to reply to the user's dev any further. We've discussed this with the customer success manager and decided to relay this to the devs as github issue.
+ HS-187518
+ This issue cannot be reproduced.
+ No one on our team can access the website (non-US IPs are blocked and US IPs hit a 502 error)
+ The user will not provide admin logins 
+ The complaint: The whole front end of the site is "extremely slow and unresponsive" when LifterLMS is enabled (LifterLMS and Non-LifterLMS pages are affected).
+ It is slightly better with _only_ LifterLMS and no other plugins enabled
+ The user claims there are no PHP or Javascript Errors
+ The user claims there are  no CPU bottlenecks, no memory issues, and that their server seems fine
+ The user hosts their own servers where they say they already use Object Caching and CDN offload
+ They've said their server has 6 cores, and 10GB of ram

### System and Environment Information

<details>
<summary>System Report</summary>

```

Home Url: [removed]

Site Url: [removed]

Login Url: [removed]/wp-login.php

Version: 5.9.1

Debug Mode: No

Debug Log: No

Debug Display: Yes

Locale: en_US

Multisite: No

Page For Posts: Not Set

Page On Front:(#24577) [[removed]/]

Permalink Structure: /%postname%/

Show On Front: page

Wp Cron: Yes

Settings

-------------------------------------------

Version: 5.10.0

Db Version: 5.10.0

Course Catalog: Course Catalog (#115) [[removed]/courses/]

Membership Catalog: My Courses (#118) [[removed]/my-courses/]

Student Dashboard: My Courses (#118) [[removed]/my-courses/]

Checkout Page: Purchase (#117) [[removed]/purchase/]

Course Catalog Per Page: 9

Course Catalog Sorting: menu_order,ASC

Membership Catalog Per Page: 9

Membership Catalog Sorting: menu_order,ASC

Site Membership: Not Set

Courses Endpoint: my-courses

Edit Endpoint: edit-account

Lost Password Endpoint: lost-password

Vouchers Endpoint: redeem-voucher

Autogenerate Username: no

Password Strength Meter: yes

Minimum Password Strength: strong

Terms Required: no

Terms Page: Not Set

Checkout Names: required

Checkout Address: optional

Checkout Phone: optional

Checkout Email Confirmation: yes

Open Registration: no

Registration Names: required

Registration Address: optional

Registration Phone: hidden

Registration Voucher: optional

Registration Email Confirmation: no

Account Names: required

Account Address: optional

Account Phone: optional

Account Email Confirmation: yes

Confirmation Endpoint: confirm-payment

Force Ssl Checkout: yes

Country: US

Currency: USD

Currency Position: left

Thousand Separator: ,

Decimal Separator: .

Decimals: 2

Trim Zero Decimals: no

Recurring Payments: no

Email From Address: [removed]

Email From Name: [removed]

Email Footer Text:

Email Header Image:

Cert Bg Width: 800

Cert Bg Height: 616

Cert Legacy Compat: no

Constants

-------------------------------------------

LLMS_REMOVE_ALL_DATA: undefined

LLMS_REST_DISABLE: undefined

LLMS_SITE_FEATURE_RECURRING_PAYMENTS: undefined

LLMS_SITE_IS_CLONE: undefined

Gateways

-------------------------------------------

Manual: Enabled

Manual Logging: no

Manual Order: 1

Server

-------------------------------------------

Mysql Version: 8.0.27

Php Curl: Yes

Php Default Timezone: UTC

Php Fsockopen: Yes

Php Max Input Vars: 1000

Php Max Upload Size: 8 MB

Php Memory Limit: 1000M

Php Post Max Size: 8M

Php Soap: Yes

Php Suhosin: No

Php Time Limt: 300

Php Version: 7.4.13

Software: Microsoft-IIS/10.0

Wp Memory Limit: 500M

Browser

-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:97.0) Gecko/20100101 Firefox/97.0

Theme

-------------------------------------------

Name: LifterLMS LaunchPad

Version: 2.4.9

Themeuri: https://lifterlms.com/launchpad/

Authoruri: https://gocodebox.com

Template:

Child Theme: No

Llms Support: No

Plugins

-------------------------------------------

Active Directory Integration for Intranet sites: 3.7.1

Colorlib Login Customizer: 1.3.1

Date Picker in List Fields for Gravity Forms: 1.7.6

Drop Down List Field for Gravity Forms: 1.8.5

Duo Two-Factor Authentication: 2.5.7

Easy WP SMTP: 1.4.7

Forms: 3rd-Party Integration: 1.8

GP Nested Forms: 1.0.7

GP Populate Anything: 1.2.2

Gravity Flow: 2.7.8

Gravity Forms: 2.5.16

Gravity Forms Encrypted Fields: 5.9

Gravity Forms Partial Entries Add-On: 1.6.1

Gravity Forms Polls Add-On: 3.9

Gravity Forms Quiz Add-On: 3.8

Gravity Forms Signature Add-On: 4.2

Gravity Forms Sticky List: 1.5.2

Gravity Forms Survey Add-On: 3.7

Gravity Forms Webhooks Add-On: 1.5

Gravity PDF: 6.1.1

Gravity Perks: 2.2.7

Hide Admin Bar From Non-admins: 1.0

If Menu - Visibility control for menu items: 0.16.2

Image Hopper Editor: 2.3.1

Image Hopper for Gravity Forms: 2.2.7

Insert Headers and Footers: 1.6.0

LifterLMS: 5.10.0

LoginWP (Formerly Peter's Login Redirect): 3.0.0.7

Members: 3.1.7

Redirection: 5.2.3

Sucuri Security - Auditing, Malware Scanner and Hardening: 1.8.30

Integrations

-------------------------------------------

BbPress: No

BuddyPress: No

Template Overrides

-------------------------------------------

```

</details>

This issue has be recreated:
+ [ ] Locally
+ [ ] On a staging site
+ [ ] On a production website
+ [ ] With only LifterLMS and a default theme

### Browser, Device, and Operating System Information

+ Browser name and version
+ Operating System name and version
+ Device name and version (if applicable)

**Comment by thomasplevy** [comment]
> The user hosts their own servers where they say they already use Object Caching and CDN offload
They've said their server has 6 cores, and 10GB of ram

The most expensive / powerful servers in the world matter not one bit when they're not configured properly.

I can speak from personal experience that when we used to self-host LifterLMS.com on AWS EC2 instances we increased CPU cores and RAM every few weeks and continued to have performance issues until we switched over to a WordPress host. What I learned was that *I AM NOT A SYSTEMS ADMINISTRATOR* and I don't know how to configure servers.

After switching to a WordPress-specific host (WPEngine) our performance issues disappeared.

---

Without the ability to physically debug this issue and see what's causing the slowness there's not much we can do. Here's some places to look:

+ Assuming everything is actually well configured by a sysadmin that understands WordPress and server administration I'd suggest having a look at the database server and ensuring the database server isn't under-powered.
+ Review database logs, especially slow query logs, and see if there's anything on the database side causing issues.
+ If traffic is being blocked, that means there's something doing the blocking. What's doing that? Is that inadvertently causing load issues?
+ What's the site's time to first byte? Perhaps there's something with the CDN causing some bottlenecks there.
